### ISSUE TITLE
Making resolutions more IDE friendly

### ISSUE THREAD

**Issue by bandophahita** [ScreenPyHQ/screenpy#11]
I was writing a test and noticed my IDE nor pylint caught a bug in my code.  Here is the simplest form:

```
re_ = ReadsExactly()
```
It wasn't until I ran it that the bug became known and even then it was initially a little cryptic.
```
Traceback (most recent call last):
  File "/Users/m.wilson/projects/screenpy/tests/test_resolutions.py", line 277, in test_can_be_instantiated
    re_ = ReadsExactly()
  File "/Users/m.wilson/projects/screenpy/screenpy/resolutions/base_resolution.py", line 84, in __init__
    self.matcher = cls.matcher_function()
TypeError: has_string() missing 1 required positional argument: 'match'
```

**Comment by bandophahita** [comment]
My IDE (pycharm) and pylint don't catch this bug because there is no constructor which can tell them of a missing argument.  

I propose we add the constructors to the resolutions.

```python
class ReadsExactly(BaseResolution):
    line = '"{expectation}", verbatim'
    matcher_function = has_string

    def __init__(self, match: str) -> None:
        super().__init__(match)
```

**Comment by perrygoy** [comment]
Closed by #12!
